core/src/main/java/org/keycloak/sdjwt/consumer/JwtVcMetadataTrustedSdJwtIssuer.java

```diff
@@ -0,0 +1,190 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A trusted Issuer for running SD-JWT VP verification.
+ *
+ * <p>
+ * This implementation targets issuers exposing verifying keys on a normalized JWT VC Issuer metadata endpoint.
+ * </p>
+ *
+ * @author <a href="mailto:Ingrid.Kamga@adorsys.com">Ingrid Kamga</a>
+ * @see <a href="https://datatracker.ietf.org/doc/html/draft-ietf-oauth-sd-jwt-vc-05#name-issuer-signed-jwt-verificat">
+ * JWT VC Issuer Metadata
+ * </a>
+ */
+public class JwtVcMetadataTrustedSdJwtIssuer implements TrustedSdJwtIssuer {
+
+    private static final String JWT_VC_ISSUER_END_POINT = "/.well-known/jwt-vc-issuer";
+
+    private final Pattern issuerUriPattern;
+    private final HttpDataFetcher httpDataFetcher;
+
+    /**
+     * @param issuerUri a trusted issuer URI
+     */
+    public JwtVcMetadataTrustedSdJwtIssuer(String issuerUri, HttpDataFetcher httpDataFetcher) {
+        try {
+            validateHttpsIssuerUri(issuerUri);
+        } catch (VerificationException e) {
+            throw new IllegalArgumentException(e);
+        }
+
+        // Build a Regex pattern to only match the argument URI
+        this.issuerUriPattern = Pattern.compile(Pattern.quote(issuerUri));
+
+        // Assign HttpDataFetcher implementation
+        this.httpDataFetcher = httpDataFetcher;
+    }
+
+    /**
+     * @param issuerUriPattern a regex pattern for trusted issuer URIs
+     */
+    public JwtVcMetadataTrustedSdJwtIssuer(Pattern issuerUriPattern, HttpDataFetcher httpDataFetcher) {
+        this.issuerUriPattern = issuerUriPattern;
+        this.httpDataFetcher = httpDataFetcher;
+    }
+
+    @Override
+    public List<SignatureVerifierContext> resolveIssuerVerifyingKeys(IssuerSignedJWT issuerSignedJWT)
+            throws VerificationException {
+        // Read iss (claim) and kid (header)
+        String iss = Optional.ofNullable(issuerSignedJWT.getPayload().get("iss"))
+                .map(JsonNode::asText)
+                .orElse("");
+        String kid = issuerSignedJWT.getHeader().getKeyId();
+
+        // Match the read iss claim against the trusted pattern
+        Matcher matcher = issuerUriPattern.matcher(iss);
+        if (!matcher.matches()) {
+            throw new VerificationException(String.format(
+                    "Unexpected Issuer URI claim. Expected=/%s/, Got=%s",
+                    issuerUriPattern.pattern(), iss
+            ));
+        }
+
+        // As per specs, only HTTPS URIs are supported
+        validateHttpsIssuerUri(iss);
+
+        // Fetch and collect exposed JWKs
+        List<JsonNode> jwks = new ArrayList<>();
+        for (JsonNode jwk : fetchIssuerMetadata(iss)) {
+            jwks.add(jwk);
+        }
+
+        // If kid specified, only consider the first matching key
+        if (kid != null) {
+            jwks = jwks.stream()
+                    .filter(jwk -> {
+                        JsonNode jwkKid = jwk.get("kid");
```

Comment:
"kid" is a magic constant

Reply:
Thank you for your remark. To avoid the incessant use of magic constants in this file, I refactored it to use existing and introduced POJOs for the data manipulated.